nf_class/components/create.py

```diff
@@ -0,0 +1,265 @@
+import logging
+import re
+from pathlib import Path
+from typing import Optional
+
+import jinja2
+import questionary
+import requests
+import yaml
+
+import nf_class
+
+# import nf_core.components.components_utils
+import nf_core.components.create
+
+# TODO: include this once new version of nf-core is released
+# import nf_core.pipelines.lint_utils
+import nf_core.lint_utils
+import nf_core.modules.modules_repo
+
+log = logging.getLogger(__name__)
+
+
+class ComponentCreateFromClass(nf_core.components.create.ComponentCreate):
+    """
+    Create a new module or subworkflow from a class.
+
+    Args:
+        ctx (dict): Click context object.
+        component_type (str): Type of component to create. [modules|subworkflows]
+        directory (str): Directory to create the component in. [default: <current directory>]
+        classname (str): Name of the class to create the component from.
+        component (str): Name of the component to create.
+        author (str): Author of the component.
+        force (bool): Overwrite existing files.
+        conda_name (str): Name of the conda environment.
+        conda_version (str): Version of the conda environment.
+
+    Attributes:
+        modules_repo (nf_core.modules.modules_repo.ModulesRepo): Modules repository object.
+        classname (str): Name of the class to create the component from.
+        inputs_yml (str): Inputs of the class in yaml format.
+        outputs_yml (str): Outputs of the class in yaml format.
+        inputs (str): Inputs of the class in string format.
+        outputs (str): Outputs of the class in string format.
+        input_vars (list): List of input variables.
+        output_vars (list): List of output variables.
+
+    Raises:
+        UserWarning: If trying to create a components for a pipeline instead of a modules repository.
```

Comment:
```suggestion
        UserWarning: If trying to create a components for a pipeline instead of a modules repository.
        UserWarning: If the required class name doesn't exist.
```